tools/python_api/src_py/query_result.py

```diff
@@ -166,7 +171,13 @@ def get_as_arrow(self, chunk_size: int) -> pa.Table:
         """
         self.check_for_query_result_close()
 
-        return self._query_result.getAsArrow(chunk_size)
+        if chunk_size <= 0 or chunk_size is None:
+            # adaptive chunk_size; target number of elements per chunk_zise
+            target_chunk_size = max(chunk_size // len(self.get_column_names()), 10)
```

Comment:
Since in this case `chunk_size` is set to `0` / `-1` / `None`, this is not useful here, as `chunk_size // len(self.get_column_names())` will always be something invalid, so this is equivalent to `target_chunk_size=10`

Reply:
Yeah the test is failing in any case because None can't be compared against an integer. Will fix.